packages/devextreme/js/__internal/ui/calendar/m_calendar.ts

```diff
@@ -744,7 +745,16 @@ const Calendar = Editor.inherit({
     }
   },
 
-  _setAriaReadonly: noop,
+  _setAriaReadonly(readonly = false) {
```

Comment:
I don't think you should use inheritance in this case. In your method you are dealing with aria-attribute logic (role + aria-label). Write a new method, for example with the name _updateAria. Handle the runtime change of the readOnly option. I see that due to inheritance, the _setAriaReadonly method runs. But I'm sure it must be responsible for other logic, judging by the name.

Reply:
[212f8b5](https://github.com/DevExpress/DevExtreme/pull/27898/commits/212f8b5f95a5f96df7644e0b287434c920ee75dc)